### DISCUSSION TITLE
Embedable FerretDB

### DISCUSSION THREAD

**Discussion by Dragomir-Ivanov** [FerretDB/FerretDB#733] [General]
Hi there,
I know that currently FerretDB is a standalone proxy, but are there any plan, to provide it as embedded package. So I can build a single Go artifact, that internally does its business logic with MongoDB protocol, but there is a FerretDB layer that outputs PostreSQL protocol to the outer world.

**Comment by AlekSi** [comment]
Hi, Dragomir. How do you want to interact with that embedded proxy? Via a MongoDB driver over a local port of Unix socket, or by some other means?

**Comment by Dragomir-Ivanov** [comment]
Hmm, that is a good question. Ideally I would like to use the official
MongoDB/Golang driver -> (magic happens) -> the app talks to PostgreSQL.
Now thinking about it, it might not be that easy. It involves hacking the
MongoDB driver, and tapping to its transport layer. I would investigate.

Thank you for your reply!

**Comment by AlekSi** [comment]
That should not require changes in the driver, as far as I can tell.

I created #3 and #737 to track the implementation. Both issues are not planned yet.

**Comment by AlekSi** [comment]
https://www.ferretdb.io/0-5-0-release-is-out-embedding-ferretdb-into-go-programs/
